ksql-serde/src/main/java/io/confluent/ksql/serde/util/SerdeUtils.java

```diff
@@ -87,4 +92,80 @@ public static double toDouble(final Object object) {
     }
     throw new IllegalArgumentException("This Object doesn't represent a double");
   }
+
+  public static boolean isCoercible(final Object object, final Schema targetSchema) {
+    Objects.requireNonNull(object, "object");
+
+    switch (targetSchema.type()) {
+      case BOOLEAN:
+        return object instanceof Boolean;
+      case INT32:
+      case INT64:
+      case FLOAT64:
+        return object instanceof Number || object instanceof String;
+      case STRING:
+        return true;
+      case ARRAY:
+        return isArrayCoercible(object, targetSchema);
+      case MAP:
+        return isMapCoercible(object, targetSchema);
+      case STRUCT:
+        return isStrictCoercible(object, targetSchema);
+      default:
+        throw new UnsupportedOperationException("Unsupported type: " + targetSchema.type());
+    }
+  }
+
+  private static boolean isArrayCoercible(final Object object, final Schema targetSchema) {
```

Comment:
nit: it feels weird that we don't check that targetSchema is an array (which I know we do in the caller). Might improve readability to just make this parameter the value schema (same comment for map/struct)

Reply:
Good call - done.